src/Microsoft.Diagnostics.EventFlow.Core/Implementations/JsonConverters/IPEndpointConverter.cs

```diff
@@ -0,0 +1,38 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+using System;
+using System.Net;
+#if NET451 || NETSTANDARD1_6
+using System.Reflection;
+#endif
+using Newtonsoft.Json;
+
+namespace Microsoft.Diagnostics.EventFlow.JsonConverters
+{
+    internal class IPEndpointConverter : JsonConverter
+    {
+        public override bool CanConvert(Type objectType)
+        {
+#if NET451 || NETSTANDARD1_6
+            return typeof(EndPoint).GetTypeInfo().IsAssignableFrom(objectType);
+#else
+            return typeof(EndPoint).IsAssignableFrom(objectType);
```

Comment:
Note:
There are 3 implementations of `EndPoint` and all of them override `.ToString()`.